client/src/main/java/cloud/prefab/client/internal/ConfigClientImpl.java

```diff
@@ -121,7 +134,30 @@ public ConfigClientImpl(
 
     if (options.isLocalOnly()) {
       finishInit(Source.LOCAL_ONLY);
+      httpClient = null;
+      connectivityTester = null;
+      prefabHttpClient = null;
     } else {
+      httpClient =
+        HttpClient
+          .newBuilder()
+          .executor(
+            MoreExecutors.getExitingExecutorService(
+              (ThreadPoolExecutor) Executors.newCachedThreadPool(
+                new ThreadFactoryBuilder()
+                  .setDaemon(true)
+                  .setNameFormat("prefab-http-client-pooled-thread-%d")
+                  .build()
+              )
+            )
+          )
+          .build();
+
+      connectivityTester =
+        new ConnectivityTester(() -> greetingServiceFutureStub(), httpClient, options);
+      connectivityTester.testHttps();
+      grpcAvailable.set(options.isGrpcEnabled() && connectivityTester.testGrpc());
```

Comment:
will this block until grpc is 👍 👎  ? 

I think it's important that we get the config unlocked asap, by whatever means are working. 

So if the CDN is working, great, unlock the configstore.  This testGrpc should block the grpc strategies, but not the others.

Reply:
no this is a one shot test.